src/audio/copier/copier.c

```diff
@@ -1446,6 +1449,38 @@ static int copier_set_sink_fmt(struct comp_dev *dev, const void *data,
 							      &sink_fmt->sink_fmt, ipc4_gtw_none,
 							      ipc4_bidirection);
 
+	/* update corresponding sink format */
+	list_for_item(sink_list, &dev->bsink_list) {
+		int i, sink_id;
+
+		sink = container_of(sink_list, struct comp_buffer, source_list);
+		sink_c = buffer_acquire(sink);
+
+		/* only updated current used sink buffer */
+		sink_id = IPC4_SINK_QUEUE_ID(sink_c->id);
+		if (sink_id != sink_fmt->sink_id) {
+			buffer_release(sink_c);
+			continue;
+		}
+
+		sink_c->stream.channels = cd->out_fmt[sink_id].channels_count;
+		sink_c->stream.rate = cd->out_fmt[sink_id].sampling_frequency;
+		audio_stream_fmt_conversion(cd->out_fmt[sink_id].depth,
+					    cd->out_fmt[sink_id].valid_bit_depth,
+					    &sink_c->stream.frame_fmt,
+					    &sink_c->stream.valid_sample_fmt,
+					    cd->out_fmt[sink_id].s_type);
+
+		sink_c->buffer_fmt = cd->out_fmt[sink_id].interleaving_style;
+
+		for (i = 0; i < SOF_IPC_MAX_CHANNELS; i++)
+			sink_c->chmap[i] = (cd->out_fmt[sink_id].ch_map >> i * 4) & 0xf;
```

Comment:
what is this?

Reply:
this is for channel map. Copier can set channel map for each output pin